### PR TITLE
USX Share VM + Share Repo

### PR DESCRIPTION
**Pull request by myothiha09**
Shared Link VM

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @myothiha09 has signed the CLA. Thanks for the pull request!

**Comment by myothiha09** [comment]
This PR is mostly reviewed here already.

https://github.com/myothiha09/box-android-share-sdk/pull/7

File: box-share-sdk/src/main/java/com/box/androidsdk/share/usx/fragments/InviteCollaboratorsFragment.java [hunk 1]
```diff
@@ -157,18 +159,18 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
             //need to log Exception
             BoxLogUtils.e(InviteCollaboratorsFragment.class.getName(), "Fetch roles request failed",
                     presenter.getException());
-            showToast(getString(presenter.getStrCode())); //was just collaborationfragment
+            showToast(getString(presenter.getStrCode()));
         }
     };
 
     private Observer<PresenterData<BoxIteratorInvitees>> onInviteesChanged = presenter -> {
```
Comment (nyjalusc) [inline box-share-sdk/src/main/java/com/box/androidsdk/share/usx/fragments/InviteCollaboratorsFragment.java]: Could you rename `presenter` to `presenterData` or `uiData` or something similar. Presenter in Android has a different meaning, I don't want the reader to get confused. If you are interested to know more, read about MVP (Model-View-Presenter) pattern.

File: box-share-sdk/src/main/java/com/box/androidsdk/share/usx/fragments/BoxFragment.java [hunk 2]
```diff
@@ -219,7 +219,22 @@ protected Spannable createTitledSpannable(final String title, final String descr
...
+        StringBuilder sb = new StringBuilder();
+        for(String curr: str.split(" ")) {
+            sb.append(Character.toUpperCase(curr.charAt(0)) + curr.substring(1) + " ");
+        }
+        sb.setLength(sb.length() - 1);
+        return sb.toString();
+    }
+
+    /**
+     * Implement this and change title through using ActionBarTitleVM.
```
Comment (nyjalusc) [inline box-share-sdk/src/main/java/com/box/androidsdk/share/usx/fragments/BoxFragment.java]: ```Implement this and change title and subtitle through ActionBarTitleVM.```

File: box-share-sdk/src/main/java/com/box/androidsdk/share/usx/fragments/CollaboratorsRolesFragment.java [hunk 3]
```diff
@@ -1,64 +1,103 @@
...
-        return -1;
+    private void showRemoveWarning() {
+        String deleteDifferentWarning = getResources().getString(R.string.box_sharesdk_warn_remove_different_collaboration_folder, vm.getName(),vm.getCollaboration().getItem().getName());
+        AlertDialog dialog = new AlertDialog.Builder(getActivity()).setTitle(R.string.box_sharesdk_title_remove_different_collaboration_folder)
+                .setMessage(deleteDifferentWarning)
+                .setPositiveButton(android.R.string.yes, new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(final DialogInterface dialog, final int which) {
+                        vm.setRemoveSelected(true);
+                        getActivity().onBackPressed();
```
Comment (nyjalusc) [inline box-share-sdk/src/main/java/com/box/androidsdk/share/usx/fragments/CollaboratorsRolesFragment.java]: If you are just trying to dismiss the alert then you should call `dialog.dismiss()` or `dialog.cancel()`

File: box-share-sdk/src/main/java/com/box/androidsdk/share/vm/SelectRoleShareVM.java [hunk 4]
```diff
@@ -21,10 +18,12 @@
     private boolean mAllowOwnerRole = false;
     private MutableLiveData<BoxCollaboration.Role> mSelectedRole = new MutableLiveData<>();
     private boolean mAllowRemove = false;
+    private boolean mRemoveSelected = false;
     private BoxCollaboration mCollaboration;
 
     private MutableLiveData<Boolean> mSendInvitationEnabled = new MutableLiveData<>();
     private MutableLiveData<Boolean> mShowSend = new MutableLiveData<>();
+    String mName = "";
```
Comment (nyjalusc) [inline box-share-sdk/src/main/java/com/box/androidsdk/share/vm/SelectRoleShareVM.java]: private?

File: box-share-sdk/src/main/java/com/box/androidsdk/share/usx/fragments/CollaboratorsRolesFragment.java (hunk 3 above)
Comment (myothiha09) [reply box-share-sdk/src/main/java/com/box/androidsdk/share/usx/fragments/CollaboratorsRolesFragment.java]: I am not only dismissing the dialog. I am also switching back to the previous fragment.

**Pull request merged**